src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2730,10 +2730,15 @@ private ConstructorDeclarationSyntax ParseConstructorDeclaration(string typeName
                 }
 
                 BlockSyntax body;
+                ArrowExpressionClauseSyntax expressionBody;
                 SyntaxToken semicolon;
-                this.ParseBodyOrSemicolon(out body, out semicolon);
-
-                return _syntaxFactory.ConstructorDeclaration(attributes, modifiers.ToTokenList(), name, paramList, initializer, body, semicolon);
+                this.ParseBlockAndExpressionBodiesWithSemicolon(out body, out expressionBody, out semicolon);
+                if (expressionBody != null)
+                {
+                    expressionBody = CheckFeatureAvailability(expressionBody, MessageID.IDS_FeatureExpressionBodiedDeOrConstructor);
```

Comment:
I put the check in `ParseBlockAndExpressionBodiesWithSemicolon` as you requested, with an assert for the meaningful IDS_xxx constants. Also I removed the code for `isAccessorBody`, because it doesn't make real sense in this context anymore.